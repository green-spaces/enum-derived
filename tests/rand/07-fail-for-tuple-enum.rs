use enum_derived::Rand;

#[derive(Rand)]
pub enum Messages {
    Email,
    Letter(String),
    Bil,
}

fn main() {}

