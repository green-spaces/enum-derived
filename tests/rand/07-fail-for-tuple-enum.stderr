error[E0277]: the trait bound `Standard: Distribution<String>` is not satisfied
 --> tests/rand/07-fail-for-tuple-enum.rs:6:12
  |
6 |     Letter(String),
  |            ^^^^^^ the trait `Distribution<String>` is not implemented for `Standard`
  |
  = help: the following other types implement trait `Distribution<T>`:
            <Standard as Distribution<()>>
            <Standard as Distribution<(A, B)>>
            <Standard as Distribution<(A, B, C)>>
            <Standard as Distribution<(A, B, C, D)>>
            <Standard as Distribution<(A, B, C, D, E)>>
            <Standard as Distribution<(A, B, C, D, E, F)>>
            <Standard as Distribution<(A, B, C, D, E, F, G)>>
            <Standard as Distribution<(A, B, C, D, E, F, G, H)>>
          and $N others
note: required by a bound in `random`
 --> $CARGO/rand-0.8.5/src/lib.rs
  |
  | where Standard: Distribution<T> {
  |                 ^^^^^^^^^^^^^^^ required by this bound in `random`
